### ISSUE TITLE
java.lang.Throwable: Read access is allowed from inside read-action (or EDT) only

### ISSUE THREAD

**Issue by Macweese** [jshiell/checkstyle-idea#595]
IDE internal error happened on startup.

OS: Windows 10 (10.0.19044.2251)
IDE: IntelliJ IDEA 2022.1.3 CE
IDE Build #IC-221.5921.22
Runtime 11.0.15+10-b2043.56 amd64
Plugin version 5.73.0

Stacktrace

```
Details: Current thread: Thread[ApplicationImpl pooled thread 7,4,main] 722868412
Dispatch thread: false; isDispatchThread(): false
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 2104291850

java.lang.Throwable: Read access is allowed from inside read-action (or EDT) only (see com.intellij.openapi.application.Application.runReadAction())
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:200)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertReadAccessAllowed(ApplicationImpl.java:1048)
	at com.intellij.openapi.roots.impl.RootIndex.<init>(RootIndex.java:63)
	at com.intellij.openapi.roots.impl.RootIndex.<init>(RootIndex.java:56)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getRootIndex(DirectoryIndexImpl.java:172)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getRootIndex(DirectoryIndexImpl.java:155)
	at com.intellij.openapi.roots.impl.DirectoryIndexImpl.getInfoForFile(DirectoryIndexImpl.java:182)
	at com.intellij.openapi.roots.impl.FileIndexBase.getInfoForFileOrDirectory(FileIndexBase.java:96)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.getInfoForFileOrDirectory(ProjectFileIndexImpl.java:27)
	at com.intellij.openapi.roots.impl.ProjectFileIndexImpl.isExcluded(ProjectFileIndexImpl.java:81)
	at com.intellij.psi.search.scope.ProjectFilesScope$1.contains(ProjectFilesScope.java:27)
	at com.intellij.psi.search.scope.packageSet.FilteredPackageSet.contains(FilteredPackageSet.java:22)
	at com.intellij.psi.search.scope.packageSet.PackageSetBase.contains(PackageSetBase.java:27)
	at org.infernus.idea.checkstyle.model.NamedScopeHelper.isFileInScope(NamedScopeHelper.java:56)
	at org.infernus.idea.checkstyle.checker.CheckStyleChecker.lambda$scan$0(CheckStyleChecker.java:51)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.infernus.idea.checkstyle.checker.CheckStyleChecker.lambda$scan$1(CheckStyleChecker.java:52)
	at java.base/java.util.Optional.map(Optional.java:265)
	at org.infernus.idea.checkstyle.checker.CheckStyleChecker.scan(CheckStyleChecker.java:50)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$inspectFile$1(CheckStyleInspection.java:135)
	at java.base/java.util.Optional.map(Optional.java:265)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$inspectFile$3(CheckStyleInspection.java:135)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.infernus.idea.checkstyle.CheckStyleInspection.inspectFile(CheckStyleInspection.java:142)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$checkFile$0(CheckStyleInspection.java:86)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.call(ApplicationImpl.java:326)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

**Comment by jshiell** [comment]
Thanks for the report - I'll have a look when I get a tick, which is likely to be mid-December thanks to work.

**Comment by jshiell** [comment]
Fixed in 5.74.1 🤞  - cheers!
